From acf94f3de1060fad3fc8434ccf12b66835cfba83 Mon Sep 17 00:00:00 2001
From: Daniel Mack <zonque@gmail.com>
Date: Fri, 7 Mar 2014 11:41:18 +0100
Subject: core/busname: add lookup string for
 BUSNAME_FAILURE_SERVICE_FAILED_PERMANENT

When a busname unit enters BUSNAME_FAILURE_SERVICE_FAILED_PERMANENT, the
serialization will not be able to look up the result as string via
busname_result_to_string(). This leads to an assertion trap during
daemon-reexec.

(cherry picked from commit 36d239dbdaf94ba2d96bb60ac45ecfc58624b1eb)
---
 src/core/busname.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/busname.c b/src/core/busname.c
index 4c34538c6524..237011ad4da6 100644
--- a/src/core/busname.c
+++ b/src/core/busname.c
@@ -548,6 +548,7 @@ DEFINE_STRING_TABLE_LOOKUP(busname_state, BusNameState);
 static const char* const busname_result_table[_BUSNAME_RESULT_MAX] = {
         [BUSNAME_SUCCESS] = "success",
         [BUSNAME_FAILURE_RESOURCES] = "resources",
+        [BUSNAME_FAILURE_SERVICE_FAILED_PERMANENT] = "failed-permanent",
 };
 
 DEFINE_STRING_TABLE_LOOKUP(busname_result, BusNameResult);
-- 
1.9.0

